src/typing/operators.ml

```diff
@@ -920,4 +920,39 @@ let type_unop ctx op flag e with_type p =
 				find_overload_or_make e
 			end
 		| AKUsingField _ | AKResolve _ ->
-			error "Invalid operation" p
\ No newline at end of file
+			error "Invalid operation" p
+
+let type_is ctx e t p_t p =
+	match t with
+	| CTPath tp ->
+		if tp.tparams <> [] then display_error ctx "Type parameters are not supported for the `is` operator" p_t;
+
+		let e = type_expr ctx e WithType.value in
+
+		let t = Typeload.load_type_def ctx p_t tp in
+		begin
+			match t with
+			| TAbstractDecl { a_impl = Some c; a_is = Some (arg_t,cf) } ->
+				(* abstract with @:op(v is T) -> generate static call *)
+				let e_method = Texpr.Builder.make_static_field c cf (mk_zero_range_pos p) in
+				let e = AbstractCast.cast_or_unify ctx arg_t e p in
```

Comment:
Unification errors should be caught and `Std.isOfType(v, T)` branch should be used.

Reply:
Are we sure about this tho? This will yield the "abstract cannot be used as a value" instead of an expected type error.

Reply:
I'm sure that if I define `@:op(A is B) static function checkIs(i:Int):Bool` then I don't want it to be involved for strings.